src/components/Tabs/Tab.tsx

```diff
@@ -0,0 +1,183 @@
+  font-family: ${({ theme }) => theme.typography.bodyStrong.fontFamily};
+  font-weight: ${({ theme }) => theme.typography.bodyStrong.weight};
+  letter-spacing: ${({ theme }) => theme.typography.bodyStrong.letterSpacing};
+  line-height: ${({ theme }) => theme.typography.bodyStrong.lineHeight};
+  text-case: ${({ theme }) => theme.typography.bodyStrong.textCase};
+
+  &:hover,
+  &:active,
+  &:focus {
+    text-decoration: none;
+    outline: none;
+  }
+
+  &:focus-visible {
+    outline: auto;
+  }
+
+  &[aria-selected='true'] {
+    color: ${({ theme }) => theme.colors.primary.text};
+    border-bottom-color: ${({ theme }) => theme.colors.primary.borderWeak};
+    ${StyledCounter} {
+      background-color: ${({ theme }) => theme.colors.primary.background};
+      border-color: ${({ theme }) => theme.colors.primary.background};
+    }
+  }
+
+  &[aria-disabled='false']:not(:disabled) {
+    &:hover,
+    &:focus,
+    &:active {
+      outline: none;
+      color: ${({ theme }) => theme.colors.primary.text};
+      border-bottom-color: ${({ theme }) => theme.colors.primary.borderWeak};
+      ${StyledCounter} {
+        background-color: ${({ theme }) => theme.colors.primary.background};
+        border-color: ${({ theme }) => theme.colors.primary.background};
+      }
+    }
+  }
+
+  &[aria-disabled='true'],
+  &:disabled {
+    cursor: not-allowed;
+    filter: grayscale(1) opacity(50%);
+  }
+`
+export type TabProps<T extends ElementType> = {
+  as?: T
+  badge?: ReactNode
+  children?: ReactNode
+  className?: string
+  counter?: number
+  disabled?: boolean
+  value?: string | number
+  onClick?: MouseEventHandler<HTMLElement>
+  onKeyDown?: KeyboardEventHandler<HTMLElement>
+  tooltip?: string
+} & Omit<
+  ComponentProps<T>,
+  | 'as'
+  | 'badge'
+  | 'children'
+  | 'className'
+  | 'counter'
+  | 'disabled'
+  | 'value'
+  | 'tooltip'
+  | 'role'
+>
+
+const Tab = <T extends ElementType = 'button'>(
+  {
+    as,
+    badge,
+    children,
+    className,
+    counter,
+    disabled = false,
+    value,
+    onClick,
+    onKeyDown,
+    tooltip,
+    ...props
+  }: TabProps<T>,
+  ref: ForwardedRef<HTMLElement>,
+) => {
+  const { selected, onChange } = useTabsContext()
+  const computedAs = as ?? 'button'
+  const isSelected = useMemo(
+    () => value !== undefined && selected === value,
+    [value, selected],
+  )
+
+  return (
+    <StyledTooltip text={tooltip}>
+      <StyledTabButton
+        role="tab"
+        ref={ref as unknown as Ref<HTMLButtonElement>}
+        className={className}
+        as={computedAs}
+        aria-label={value ? `${value}` : undefined}
+        aria-selected={isSelected}
+        aria-disabled={disabled}
+        disabled={computedAs === 'button' ? disabled : undefined}
+        onClick={event => {
+          if (value) {
+            onChange(value)
+          }
+          ;(onClick as MouseEventHandler<HTMLElement>)?.(event)
```

Comment:
`onClick` already has a type of `MouseEventHandler<HTMLElement> | undefined`, why do we need to cast?